src/bmp/decoder.rs

```diff
@@ -670,42 +702,73 @@ impl<R: Read + Seek> BmpDecoder<R> {
-                    ))
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
+                        "Invalid RLE8 bit count".to_owned(),
+                    )))
                 }
             },
             2 => match self.bit_count {
                 4 => ImageType::RLE4,
                 _ => {
-                    return Err(ImageError::FormatError(
-                        "Invalid RLE4 bit count".to_string(),
-                    ))
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
+                        "Invalid RLE4 bit count".to_owned(),
+                    )))
                 }
             },
             3 => match self.bit_count {
                 16 => ImageType::Bitfields16,
                 32 => ImageType::Bitfields32,
                 _ => {
-                    return Err(ImageError::FormatError(
-                        "Invalid bitfields bit count".to_string(),
-                    ))
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
+                        "Invalid bitfields bit count".to_owned(),
+                    )))
                 }
             },
-            // PNG and JPEG not implemented yet.
+            4 => {
+                // JPEG compression is not implemented yet.
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Bmp.into(),
+                        UnsupportedErrorKind::GenericFeature("JPEG compression".to_owned()),
+                    ),
+                ));
+            }
+            5 => {
+                // PNG compression is not implemented yet.
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Bmp.into(),
+                        UnsupportedErrorKind::GenericFeature("PNG compression".to_owned()),
+                    ),
+                ));
+            }
+            11 | 12 | 13 => {
+                // CMYK types are not implemented yet.
+                return Err(ImageError::Unsupported(
+                    UnsupportedError::from_format_and_kind(
+                        ImageFormat::Bmp.into(),
+                        UnsupportedErrorKind::GenericFeature("CMYK format".to_owned()),
+                    ),
+                ));
+            }
             _ => {
-                return Err(ImageError::UnsupportedError(
-                    "Unsupported image type".to_string(),
-                ))
+                // Unknown compression enumeration.
+                return Err(ImageError::Decoding(DecodingError::with_message(
+                    ImageFormat::Bmp.into(),
+                    format!("Unknown image type {}", image_type_u32),
```

Comment:
Would it be accurate to say "unknown image compression type {}"?

Reply:
Should "Unknown" be all lowercase?

Reply:
It's more consistent to have an upper case at the start, also with other errors. So no :smiley: